### ISSUE TITLE
RSS Fingerprint indoor localization using neural and SVR  algorithm 

### ISSUE THREAD

**Issue by ashrafs** [haifengl/smile#38]
Dear Sir
I use your algorithm published in smile in regression SVR And RBF neural network in my thesis for enhance indoor localization WLAN .
 and many thanks for your effort .
 I built RSS Fingerprint and implement affinity propagation algorithm for cluster data  and i used training RBF neural network in offline phase and in  online phase i used ANN estimation for test new data

paper I use it as reference at link .
https://www.dropbox.com/home?preview=06554922.pdf

my question 

in paper author used Affinity Propagation and RBF network as two phases for localization (coarse and fine), as shown in the figure below (from the paper).
![untitled](https://cloud.githubusercontent.com/assets/5106208/9713309/e905683a-5550-11e5-8693-e247d3bbcc97.png)
My question is: how I can get the two techniques to work together in the online phase? How i can use the output of phase 1 (cluster matching) as input for phase 2 (ANN estimation)?
In other words, how I can tell the RBF network that its search must be restricted to cluster X (returned from cluster matching)? 
what's your advise how I can do that?
thanks a lot

**Comment by haifengl** [comment]
The first phase of RBF networks is just k-means. If you need only that, why not just use k-means?

**Comment by ashrafs** [comment]
Kmeans  need predefined number of cluster and its one from disadvantages. 
Affinity Propogation used to cluster all data after that I use rbf neural for train all cluster  alone. 
Ii implement all one from components in graph above .
My Qz  how I can integrate this part?
Any code integrate between cluster and train all  one from cluster with rbf nn will be helpful

**Comment by haifengl** [comment]
K-means inside rbf networks still needs the number of clusters. The trainer class has a method setNumCenters(int) to set it.

**Comment by ashrafs** [comment]
I can use just one cluster algorithm
I use affinity propagation for cluster data at first .
and I can use same algorithm in rbf with predefined centroid to build gaussian 
thats correct?

**Comment by haifengl** [comment]
Yes, you can use predefined centroids. When creating a RBFNetwork instance, you can provide them in the constructor.

**Comment by ashrafs** [comment]
If I train all one from cluster in rbf neural network .
in this case all cluster have one centroid?

**Comment by haifengl** [comment]
I don't understand your question. It is not valid english. sorry.

**Comment by ashrafs** [comment]
I ask about 
after use cluster algorithm for data 
Each one will have just one centroid in this case?

**Comment by haifengl** [comment]
For many partitioning clustering algorithms (e.g. k-means), each cluster has just one centroid.

**Comment by ashrafs** [comment]
Thank you
